.github/workflows/beta.yml

```diff
@@ -0,0 +1,99 @@
+
+on:
+  workflow_dispatch:
+    inputs:
+      git-ref:
+        required: true
+        type: string
+        description: "The github ref of this beta version (i.e. main, 1234567)"
+        default: main
+      publish:
+        required: false
+        type: boolean
+        default: false
+        description: "Publish the beta release"
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  crate:
+    name: Publish surrealdb to crates.io
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install stable toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.git-ref || github.ref_name }}
+
+      - name: Install release-plz
+        run: cargo install --force --locked --version 0.3.30 release-plz
+
+      - name: Install Taplo
+        run: cargo install taplo-cli --locked
+
+      - name: Prepare beta crate
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+        run: |
+          set -x
+
+          # Configure git
+          git config --local user.email "actions@users.noreply.github.com"
+          git config --local user.name "github-actions"
+
+          # Bump the version
+          /home/runner/.cargo/bin/release-plz update --config .config/release-plz.toml
+          
+          # Save the new version
+          newVersion=$(/home/runner/.cargo/bin/taplo get -f lib/Cargo.toml "package.version")
+          version=${newVersion}-beta.1
+          
+          # Stash the Cargo.toml
+          tempDir=$(mktemp -d /tmp/cargo-toml.XXXXXXXXXXXXXXXXXX)
+          mv lib/Cargo.toml $tempDir/
+
+          # Temporarily revert the change and create the tag
+          git checkout -- lib/Cargo.toml
+          git tag -a v${version} -m "Release ${version}"
+
+          # Unstash Cargo.toml
+          mv $tempDir/Cargo.toml lib/
+
+          # Replace the crate name
+          # We are just going to replace the first occurance of surrealdb
+          sed -i "0,/surrealdb/s//surrealdb-beta/" lib/Cargo.toml
+
+          # Commit changes
+          git checkout -b beta
+          git add lib/Cargo.toml
+          git commit -m "Prepare the beta crate"
+
+      - name: Perfom release checks
+        if: ${{ !inputs.publish }}
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+        run: /home/runner/.cargo/bin/release-plz release --dry-run --config .config/release-plz.toml
+
+      - name: Publish the crate
+        if: ${{ inputs.publish }}
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+          GIT_TOKEN: ${{ secrets.RELEASE_PLZ_TOKEN }}
+        run: |
+          # Upload the crate
+          /home/runner/.cargo/bin/release-plz release --config .config/release-plz.toml
+
+          # Push the new branch and tag
```

Comment:
can we use the built-in GIT_TOKEN?

https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-1-passing-the-github_token-as-an-input

Reply:
The builtin token is known to not trigger subsequent workflows @sgirones. That's why we had to set up a new token so that the new tags will be pushed as if they were triggered by the user. The beta workflow will need to push the tag and then the release workflow will need to pick up on that.

Reply:
could we use `workflow_dispatch` to triggered the desired workflows? The argument for not triggering workflows when using the built-in token seams reasonable, so maybe it's better to not bypass that protection?

Reply:
We are already triggering this workflow manually and the release workflow will be triggered by a tag that would already have been pushed to the repo. Requiring that the release workflow be triggered manually too after the tag is pushed is unnecessary (I don't see any additional protections it adds) and error prone because someone might forget to run it after running this workflow. Besides, the release workflow already fires on a tag and that's been working well for us so far. I don't see any reason to force only running it manually.

Reply:
@sgirones Would you be more comfortable with calling `reusable_publish_version.yml` directly in the beta workflow? If we do that, we wouldn't need to create and push the beta tag.

Reply:
I personally find it more intuitive, yeah.

Is the beta tag useful otherwise? Not sure about the implications of not creating it. We may still want to have a Github release/tag for each beta so we have access to it in the future.